test/json.jl

```diff
@@ -46,6 +46,11 @@ end
     @test JSON3.read("Inf"; allow_inf=true) === Inf
     @test JSON3.read("Infinity"; allow_inf=true) === Inf
     @test JSON3.read("-Infinity"; allow_inf=true) === -Inf
```

Comment:
Noting that it is not possible to support reading with a custom mapping defined with a function that maps `Float64` to `String`.

However, it is possible to support reading and writing with a `@NamedTuple{positive_inf::AbstractString, negative_inf::AbstractString, nan::AbstractString}` API.

Reply:
Yeah, I agree that not being able to read the JSON back in when a custom mapping is used is a bummer, but also it's something that is/will be better solved in JSONBase.jl, where it's easier to override reading things.

Actually, we do have the RawJson construct if you really needed to parse something back in; it's a bit of a heavy-handed escape hatch here, but technically would work.